include/motor-control/core/motor_interrupt_handler.hpp

```diff
@@ -104,18 +104,37 @@ class MotorInterruptHandler {
             update_move();
             return false;
         }
-        if (has_active_move && can_step() && tick()) {
-            return true;
-        }
-        if (has_active_move && !can_step()) {
-            finish_current_move();
-            if (has_messages()) {
-                update_move();
-                if (can_step() && tick()) {
-                    return true;
+        if (has_active_move) {
+            if (buffered_move.stop_condition ==
+                    MoveStopCondition::limit_switch &&
+                limit_switch_triggered()) {
+                return false;
+            }
+            if (can_step() && tick()) {
+                return true;
+            }
+            if (!can_step()) {
+                finish_current_move();
+                if (has_messages()) {
+                    update_move();
+                    if (can_step() && tick()) {
+                        return true;
+                    }
                 }
+                return false;
             }
-            return false;
+        }
+        return false;
+    }
+
+    auto limit_switch_triggered() -> bool {
+        if (hardware.check_limit_switch() && can_step()) {
```

Comment:
This functionality means that
- If the limit switch is triggered on the last _tick_ of the move, there's no error
- If the limit switch is triggered on any other tick, there's an error

There's a couple problems with this. First, we're going to use this for homing - we _want_ to hit the limit switch, that's not an error. Second, checking on the last _tick_ as opposed to the last _step_ basically ensures this won't happen - the last step would be the thing that hits the switch.

Reply:
I think this all makes sense. One caveat that I think we should take into account is that if a stop condition that was specified is _not_ met, then we should consider that an error.